### PR TITLE
🐛 Fix missing signer issue during redirect for AuthCore

### PR DESCRIPTION
**Pull request by AuroraHuang22**
https://github.com/user-attachments/assets/7403a365-9c98-492b-ba93-0a93434f82dd

### CODE REVIEW

File: pages/auth/redirect.vue [hunk 1]
```diff
@@ -23,8 +31,22 @@ onMounted(async () => {
       method: method as string,
       params: { code }
     })
+
+    if (signer.value) {
```
Comment (williamchong) [inline pages/auth/redirect.vue:35]: What does it mean for `signer.value` to have no value? Should handle like error case

Comment (AuroraHuang22) [reply pages/auth/redirect.vue:36]: [🥅 Handle missing signer issue](https://github.com/likecoin/nft-book-press/pull/175/commits/ef25d8ea1b16af5de132a12f4ac683c5126fa1c8)

File: pages/auth/redirect.vue [hunk 2]
```diff
@@ -23,8 +31,22 @@ onMounted(async () => {
...
+      const signature = await signMessageMemo(
+        'authorize',
+        ['read:nftbook', 'write:nftbook', 'read:nftcollection', 'write:nftcollection']
+      )
+      if (!signature) { return }
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply pages/auth/redirect.vue]: [🥅 Handle missing signer issue](https://github.com/likecoin/nft-book-press/pull/175/commits/ef25d8ea1b16af5de132a12f4ac683c5126fa1c8)

File: pages/auth/redirect.vue [hunk 3]
```diff
@@ -23,8 +32,35 @@ onMounted(async () => {
       method: method as string,
       params: { code }
     })
+
+    if (signer.value) {
+      const signature = await signMessageMemo(
+        'authorize',
+        ['read:nftbook', 'write:nftbook', 'read:nftcollection', 'write:nftcollection']
+      )
+      if (!signature) { disconnect(); return router.replace('/') }
```
Comment (williamchong) [inline pages/auth/redirect.vue:41]:
```suggestion
      if (!signature) {
        disconnect()
        router.replace('/')
        return
      }
```
Prefer consistent return value for a function, in this case we originally return void at the end

**Comment by nwingt** [comment]
Not sure if we want to disconnect if the connected wallet changed

File: pages/auth/redirect.vue (hunk 3 above)
Comment (AuroraHuang22) [reply pages/auth/redirect.vue]: [🥅 Handle missing signer issue](https://github.com/likecoin/nft-book-press/pull/175/commits/635e5ef095ca3c85fa782a13fec9c696aa991068)

**Review by williamchong — APPROVED** [review]

**Comment by williamchong** [comment]
> Not sure if we want to disconnect if the connected wallet changed

You mean globally?

**Comment by nwingt** [comment]
> > Not sure if we want to disconnect if the connected wallet changed
> 
> You mean globally?

Yes

**Pull request merged**